pkg/schedule/schedulers/hot_region.go

```diff
@@ -1504,6 +1507,94 @@ func (bs *balanceSolver) buildOperators() (ops []*operator.Operator) {
 	return
 }
 
+// bucketFirstStat returns the first priority statistics of the bucket.
+// if the first priority is query rate, it will return the second priority .
+func (bs *balanceSolver) bucketFirstStat() statistics.RegionStatKind {
+	dim := statistics.RegionReadBytes
+	diff := bs.firstPriority
+	// todo: remove it if bucket's qps has been supported.
+	if bs.firstPriority == statistics.QueryDim {
+		diff = bs.secondPriority
+	}
+	if bs.rwTy == statistics.Write {
+		dim = statistics.RegionWriteBytes
+	}
+	dim += statistics.RegionStatKind(diff)
+	return dim
+}
+
+func (bs *balanceSolver) splitBucketsOperator(region *core.RegionInfo, keys [][]byte) *operator.Operator {
+	if len(keys) == 0 {
+		hotSchedulerNotFoundSplitKeysCounter.Inc()
+		return nil
+	}
+	des := splitHotReadBuckets
+	if bs.rwTy == statistics.Write {
+		des = splitHotWriteBuckets
+	}
+	op, err := operator.CreateSplitRegionOperator(des, region, operator.OpSplit, pdpb.CheckPolicy_USEKEY, keys)
+	if err != nil {
+		log.Error("fail to create split operator",
+			zap.Stringer("resource-type", bs.resourceTy),
+			errs.ZapError(err))
+		return nil
+	}
+	hotSchedulerSplitSuccessCounter.Inc()
+	return op
+}
+
+func (bs *balanceSolver) splitHotKeys(region *core.RegionInfo, stats []*buckets.BucketStat) *operator.Operator {
+	totalLoads := uint64(0)
+	dim := bs.bucketFirstStat()
+	for _, stat := range stats {
+		totalLoads += stat.Loads[dim]
+	}
+
+	acc, splitIdx := uint64(0), 0
+	for ; acc < totalLoads/2 && splitIdx < len(stats); splitIdx++ {
+		acc += stats[splitIdx].Loads[dim]
+	}
```

Comment:
Would you like to add more comments about acc?

Reply:
yes, fixed